src/main/java/org/apache/commons/lang3/math/NumberUtils.java

```diff
@@ -944,11 +952,15 @@ public static Long createLong(final String str) {
      * @return converted <code>BigInteger</code> (or null if the input is null)
      * @throws NumberFormatException if the value cannot be converted
      */
+    @SuppressWarnings("compound.assignment.type.incompatible")/*
+    #4 - str.startsWith("0x")=> @MinLen(2)
+    #5 - str.startsWith("#") => @MinLen(1)
```

Comment:
It probably wouldn't be too hard to implement this as a special case in the Value Checker, if you're interested.

Reply:
Though I don't actually think that would make this code compile without errors.